api/v1beta1/glance_webhook.go

```diff
@@ -105,6 +120,41 @@ func (spec *GlanceSpec) Default() {
 
 //+kubebuilder:webhook:path=/validate-glance-openstack-org-v1beta1-glance,mutating=false,failurePolicy=fail,sideEffects=None,groups=glance.openstack.org,resources=glances,verbs=create;update,versions=v1beta1,name=vglance.kb.io,admissionReviewVersions=v1
 
+// Check if File is used as a backend for Glance
+func isFileBackend(customServiceConfig string, topLevel bool) bool {
+
+	availableBackends := GetEnabledBackends(customServiceConfig)
+	// if we have "enabled_backends=backend1:type1,backend2:type2 ..
+	// we need to iterate over this list and look for type=file
+	for i := 0; i < len(availableBackends); i++ {
+		backendToken := strings.SplitN(availableBackends[i], ":", 2)
+		if (backendToken[1] == "file") {
+			return true
+		}
+	}
+	// If the iteration over the list has not produced file, we have yet another
+	// possible scenario to evaluate:
+	// - availableBackends is []
+	// - the topLevel CR is [] or has File has backend (topLevel is true)
+	if len(availableBackends) == 0 && topLevel {
+		return true
+	}
+	return false
+}
+
+// Check if the File is used in combination with a wrong layout
+func (r *Glance) isInvalidBackend(glanceAPI GlanceAPITemplate, topLevel bool) bool {
+	var rep int32 = 0
+	// For each current glanceAPI instance, detect an invalid configuration
+	// made by "type: split && backend: file": raise an issue if this config
+	// is found. However, do not fail if 'replica: 0' because it means the
+	// operator has not made any choice about the backend yet
+	if (*glanceAPI.Replicas != rep && glanceAPI.Type == "split" && isFileBackend(glanceAPI.CustomServiceConfig, topLevel)) {
```

Comment:
During the ctlplane CR creation, we do not want to fail if we set the `type` to split but at the same time `Replicas` is 0.
If we look at VA2 [1], we have this kind of situation where no backend has been chosen yet, no `customConfigService` is passed and `replicas: 0`.
We need, in that case, the `ospctlplane` to converge into a working state, and I think it makes sense evaluate a condition that takes into account the number of replicas.
@abays wdyt?

Reply:
Yes, that makes sense to me